MathlibTest/DeprecatedSyntaxLinter.lean

```diff
@@ -0,0 +1,34 @@
+import Mathlib.Tactic.Cases
+import Mathlib.Tactic.Linter.DeprecatedSyntaxLinter
+
+set_option linter.style.refine true
+/--
+warning:
+The `refine'` tactic is discouraged: please strongly consider using `refine` or `apply` instead.
+note: this linter can be disabled with `set_option linter.style.refine false`
+---
+warning:
+The `refine'` tactic is discouraged: please strongly consider using `refine` or `apply` instead.
+note: this linter can be disabled with `set_option linter.style.refine false`
+-/
+#guard_msgs in
+example : True := by
+  refine' (by refine' .intro)
+
+set_option linter.style.refine false
+set_option linter.style.cases true
+
+-- But this is now quiet as `linter.style.refine` is now false
+example : True := by
+  refine' (by refine' .intro)
+
+/--
+warning: The `cases'` tactic is discouraged: please strongly consider using `obtain`, `rcases` or `cases` instead.
+note: this linter can be disabled with `set_option linter.style.cases false`
+---
+warning: The `cases'` tactic is discouraged: please strongly consider using `obtain`, `rcases` or `cases` instead.
+note: this linter can be disabled with `set_option linter.style.cases false`
+-/
+#guard_msgs in
```

Comment:
Optional: copy this test-case, but with `linter.style.cases also false - and assert that nothing is printed.

Reply:
Done.